packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -7,26 +12,232 @@ import { IVirtualService } from './virtual-service';
 import { Construct } from '@aws-cdk/core';
 
 /**
- * The criterion for determining a request match for this GatewayRoute
+ * Configuration for gateway route host name match.
+ */
+export interface GatewayRouteHostnameMatchConfig {
+  /**
+   * GatewayRoute CFN configuration for host name match.
+   */
+  readonly hostnameMatch: CfnGatewayRoute.GatewayRouteHostnameMatchProperty;
+}
+
+/**
+ * Used to generate host name matching methods.
+ */
+export abstract class GatewayRouteHostnameMatch {
+  /**
+   * The value of the host name must match the specified value exactly.
+   *
+   * @param name The exact host name to match on
+   */
+  public static exact(name: string): GatewayRouteHostnameMatch {
+    return new GatewayRouteHostnameMatchImpl({
+      exact: name,
+    });
+  }
+
+  /**
+   * The value of the host name with the given name must end with the specified characters.
+   *
+   * @param suffix The specified ending characters of the host name to match on
+   */
+  public static endingWith(suffix: string): GatewayRouteHostnameMatch {
+    return new GatewayRouteHostnameMatchImpl({
+      suffix: suffix,
+    });
+  }
+
+  /**
+   * Returns the gateway route host name match configuration.
+   */
+  public abstract bind(scope: Construct): GatewayRouteHostnameMatchConfig;
+}
+
+class GatewayRouteHostnameMatchImpl extends GatewayRouteHostnameMatch {
+  constructor(
+    private readonly matchProperty: CfnGatewayRoute.GatewayRouteHostnameMatchProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): GatewayRouteHostnameMatchConfig {
+    return {
+      hostnameMatch: this.matchProperty,
+    };
+  }
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute.
  */
 export interface HttpGatewayRouteMatch {
   /**
-   * Specifies the path to match requests with.
-   * This parameter must always start with /, which by itself matches all requests to the virtual service name.
-   * You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
-   * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   * Specify how to match requests based on the 'path' part of their URL.
+   *
+   * @default - matches requests with any path
```

Comment:
Again, pretty sure the default should be:

```suggestion
   * @default - do not match on path
```

Reply:
This can be left as it is.